src/test/java/systems/crigges/jmpq3test/MpqTests.java

```diff
@@ -396,4 +408,27 @@ public void newBlocksizeBufferOverflow() throws IOException {
 
         mpqEditor.close();
     }
+	
+	@Test()
+	public void testForGetMpqFileByBlock() throws IOException {
+		File[] mpqs = getMpqs();
+		for (File mpq : mpqs) {
+			if (mpq.getName().equals("invalidHashSize.scx_copy")) {
+				continue;
+			}
+			try (JMpqEditor mpqEditor = new JMpqEditor(mpq, MPQOpenOption.FORCE_V0)) {
+				
+				Assert.assertTrue(mpqEditor.getMpqFilesByBlockTable().size()>0);
+				BlockTable blockTable = mpqEditor.getBlockTable();
+				Assert.assertNotNull(blockTable);
+				for (BlockTable.Block block : blockTable.getAllVaildBlocks())
+				{
+					if ((block.getFlags() & MpqFile.ENCRYPTED) == MpqFile.ENCRYPTED){
```

Comment:
use hasFlag

Reply:
this code is copy from JmpqEditer.java line 493 .
well . If you want to change, I will